### ISSUE TITLE
Support for Ruby 2.7

### ISSUE THREAD

**Issue by marcotc** [DataDog/dd-trace-rb#805]
With Ruby 2.7 [scheduled to be released December 2019](https://www.ruby-lang.org/en/news/2019/05/30/ruby-2-7-0-preview1-released/), `dd-trace-rb` should be ready to support it on release date.

There are a few confirmed blockers affecting the latest release, `2.7.0-preview1`:
#### Blockers
- `2.7.0-preview1` is affected by https://bugs.ruby-lang.org/issues/15891, which prevents `dd-trace-rb` from starting up successfully. This has been fixed in Ruby `master` branch, which is still unreleased.
#### Contrib
- `grpc`, [as of version 1.23.0](https://rubygems.org/gems/grpc/versions/1.23.0), doesn't support Ruby 2.7, failing installation with: `grpc requires Ruby version >= 2.3, < 2.7.dev`

**Comment by delner** [comment]
For `grpc` we could simply restrict the version support to < 2.7 until gRPC is updated.

**Comment by marcotc** [comment]
I created a branch to test this, after `2.7.0-preview3` has been released: https://github.com/DataDog/dd-trace-rb/tree/ci/ruby-2.7-preview3

All gems loaded properly, except for `grpc` and `graphql`: `grpc` has the same issue mentioned earlier, and `graphql` tries to load a file that has been removed in `2.7`.

For everything else, tests pass.

I'll follow up again closer to release, but things look promising so far.
